Use channel's default_auto_archive_duration for new threads

jgayfer opened the pull request:
## Summary

Currently creating a new thread uses a default auto archive duration of
1440 minutes, or 1 day.

Rather than prescribing our own default, we can use the default auto
archive duration that is set on the channel. This ensures that newly
created threads will respect the default auto archive duration as
prescribed by the user.

My initial approach was to simply omit the `auto_archive_duration` from
the request, but when doing so, the thread is always created with an
auto archive duration of 1440 minutes, regardless of the channel's
default setting.

See the following PR for more discussion on the matter:

  https://github.com/discord/discord-api-docs/pull/3567

It's unclear if this is intended behaviour or not, but we might as well
work with it for the time being

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

Rapptz reviewed (commented):
The library generally uses `MISSING` instead of `Optional[X]` for type checking purposes and because `None` is overloaded in the API to mean "not set". I think for here it looks superficially okay, except this is no longer symmetrical with `edit` (i.e. `Thread.edit(auto_archive_duration=None)` doesn't work. Note that I'm not saying to make it work, I'm just giving extra rationale why `MISSING` is the better option from a library design perspective.

jgayfer commented:
@Rapptz Thanks for the feedback! I wasn't aware of `MISSING`, but it makes sense to differentiate between values that aren't provided versus those that are intentionally blank.

I changed the default values from `None` to `MISSING`, and updated the type hints accordingly.

Rapptz reviewed (approved):
LGTM.
